Sources/Quick/QuickTestObservation.swift

```diff
@@ -12,16 +12,24 @@ import XCTest
 @objc internal final class QuickTestObservation: NSObject, XCTestObservation {
     @objc(sharedInstance)
     static let shared = QuickTestObservation()
+    
+    override init() {
+        super.init()
+        QuickSpec.enumerateSubclasses { specClass in
```

Comment:
Note that the init will be called when types are instantiated (b/c of https://github.com/Quick/Quick/pull/1124/files#diff-6288283bc6481ff91171d7266acdfb23cfb0b8d5dba40176ba0e05046cd6b5f4R14). This results in the runtime subclass enumeration being called while classes are still being loaded into runtime. 

In our codebase this resulted in random crashes happening when the tests started running (I also tried a poor mans approach by wrapping this in an async call but that was already too late).

Reply:
Could you tell me about the crash scene? A large number of case tests found no problems

Reply:
and,   
```
    override init() {
        super.init()
        
        DispatchQueue.main.async {
            QuickSpec.enumerateSubclasses { specClass in
                // This relies on `_QuickSpecInternal`.
                (specClass as AnyClass).buildExamplesIfNeeded()
            }
        }
    }
```
is it better？

Reply:
In our rather large code base (200k LoC with multiple modules) doing the init-based approach results in:

```
runtime assert: unexpected initialization order
```

This totally makes sense - this code is trying to walk through runtime while runtime is still being loaded (see screenshot below for details). 

Also as mentioned above: I tried wrapping this into an async call and that's too late - no tests are being registered 😞 

<img width="1272" alt="Screenshot 2022-03-18 at 13 14 54" src="https://user-images.githubusercontent.com/968513/159001226-e7c20818-497b-4bcc-aea1-aa3ff2c4f05f.png">